src/fmu/dataio/_metadata.py

```diff
@@ -80,6 +65,56 @@ def generate_meta_tracklog() -> list[meta.TracklogEvent]:
     ]
 
 
+def _get_objectdata_provider(
+    object: types.Inferrable,
+    dataio: ExportData,
+    meta_existing: dict | None = None,
+) -> ObjectDataProvider:
+    """Analyze the actual object together with input settings.
+
+    This will provide input to the ``data`` block of the metas but has also
+    valuable settings which are needed when providing filedata etc.
+
+    Hence this must be ran early or first.
+    """
+    objdata = objectdata_provider_factory(object, dataio, meta_existing)
+    objdata.derive_metadata()
+    return objdata
+
+
+def _get_filedata_provider(
+    dataio: ExportData, objdata: ObjectDataProvider, fmudata: FmuProvider | None
+) -> FileDataProvider:
+    filedata = FileDataProvider(
+        dataio=dataio,
+        objdata=objdata,
+        rootpath=dataio._rootpath,  # has been updated to case_path if fmurun
+        itername=fmudata.get_iter_name() if fmudata else "",
+        realname=fmudata.get_real_name() if fmudata else "",
+    )
+    filedata.derive_filedata()
+    return filedata
+
+
+def _compute_md5(
+    dataio: ExportData, objdata: ObjectDataProvider, object: types.Inferrable
+) -> str:
+    """Return the file block in the metadata."""
+
+    if not objdata.extension.startswith("."):
+        raise ValueError("An extension must start with '.'")
+    with NamedTemporaryFile(
+        buffering=0,
+        suffix=objdata.extension,
+    ) as tf:
+        logger.info("Compute MD5 sum for tmp file...: %s", tf.name)
+        return export_file_compute_checksum_md5(
+            obj=object,
+            filename=Path(tf.name),
+            flag=dataio._usefmtflag,
+        )
+
+
 @dataclass
 class MetaData:
```

Comment:
The need for a class is now a bit vague to me after removing all the internal state storing attributes.. should we consider changing to a function instead?

Reply:
Yes, I think that would be a very nice change given your changes here.